code/datums/components/area_sound_manager.dm

```diff
@@ -0,0 +1,77 @@
+///Allows you to set a theme for a set of areas without tying them to looping sounds explicitly
+/datum/component/area_sound_manager
+	///area -> looping sound type
+	var/list/area_to_looping_type = list()
+	///Current sound loop
+	var/datum/looping_sound/our_loop
+	///A list of "acceptable" z levels to be on. If you leave this, we're gonna delete ourselves
+	var/list/accepted_zs
+	///The timer id of our current start delay, if it exists
+	var/timerid
+
+/datum/component/area_sound_manager/Initialize(area_loop_pairs, change_on, remove_on, acceptable_zs)
+	if(!ismovable(parent))
+		return
+	area_to_looping_type = area_loop_pairs
+	accepted_zs = acceptable_zs
+	change_the_track()
+
+	RegisterSignal(parent, COMSIG_MOVABLE_MOVED, .proc/react_to_move)
+	RegisterSignal(parent, COMSIG_MOVABLE_Z_CHANGED, .proc/react_to_z_move)
+	RegisterSignal(parent, change_on, .proc/handle_change)
+	RegisterSignal(parent, remove_on, .proc/handle_removal)
+
+/datum/component/area_sound_manager/Destroy(force, silent)
+	QDEL_NULL(our_loop)
+	. = ..()
+
+/datum/component/area_sound_manager/proc/react_to_move(datum/source, atom/oldloc, dir, forced)
+	SIGNAL_HANDLER
+	var/list/loop_lookup = area_to_looping_type
+	if(loop_lookup[get_area(oldloc)] == loop_lookup[get_area(parent)])
+		return
+	change_the_track(TRUE)
+
+/datum/component/area_sound_manager/proc/react_to_z_move(datum/source, old_z, new_z)
+	SIGNAL_HANDLER
+	if(!length(accepted_zs) || (new_z in accepted_zs))
+		return
+	qdel(src)
+
+/datum/component/area_sound_manager/proc/handle_removal(datum/source)
+	SIGNAL_HANDLER
+	qdel(src)
+
+/datum/component/area_sound_manager/proc/handle_change(datum/source)
+	SIGNAL_HANDLER
+	change_the_track()
+
+/datum/component/area_sound_manager/proc/change_the_track(skip_start = FALSE)
+	var/time_remaining = 0
+
+	if(our_loop)
+		var/our_id = our_loop.timerid || timerid
+		if(our_id)
+			time_remaining = timeleft(our_id, SSsound_loops) || 0
```

Comment:
this bit of code relies on a short circuit. So it's not if either is true use that, it's more `if the first is true, use that, otherwise use the second.`
![image](https://user-images.githubusercontent.com/58055496/119416691-1273b580-bca9-11eb-900f-0972a3115023.png)
Now is this good practice? that's another question entirely

Reply:
yes, it is

Reply:
well I mean he had to ask, so it's clearly not so clear